osgp/platform/osgp-secret-management/src/main/java/org/opensmartgridplatform/secretmanagement/application/services/SecretManagementService.java

```diff
@@ -247,7 +252,70 @@ private EncryptedTypedSecret retrieveSecret(
     return Optional.of(secretsList.iterator().next());
   }
 
-  public synchronized void storeSecrets(
+  public synchronized void storeSingleNewSecrets(
+      final String deviceIdentification, final List<TypedSecret> secrets) {
+    final List<TypedSecret> typedSecretsToStore = new ArrayList<>();
+    final List<TypedSecret> typedSecretsToReset = new ArrayList<>();
+    for (final TypedSecret typedSecret : secrets) {
+      final int numberOfNewSecrets =
+          this.secretRepository.getSecretCount(
+              deviceIdentification, typedSecret.getSecretType(), SecretStatus.NEW);
+      if (numberOfNewSecrets == 0) {
+        typedSecretsToStore.add(typedSecret);
+      } else if (this.recentNewSecretsPresent(
+          deviceIdentification, this.maxMinutesForNewKeyToBeActivated, typedSecret)) {
+        final String errorMsg =
+            "There is/are secrets of type %s for device %s with status NEW "
+                + "created less than %d minutes old. No key with status NEW will be stored. Wait "
```

Comment:
`less than %d minutes old` or `created less than %d minutes ago`

Reply:
resolved the _contamination_